src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalCluster.java

```diff
@@ -18,97 +18,38 @@
  *
  * @author Michael J. Simons
  */
 class CausalCluster implements CloseableResource {
 
-	private static final int DEFAULT_BOLT_PORT = 7687;
-
-	static CausalCluster start(Configuration configuration) {
-
-		final int numberOfCoreMembers = configuration.getNumberOfCoreMembers();
-
-		// Prepare one shared network for those containers
-		final Network network = Network.newNetwork();
-
-		// Setup a naming strategy and the initial discovery members
-		final String initialDiscoveryMembers = configuration.iterateCoreMembers()
-			.map(n -> String.format("%s:5000", n.getValue()))
-			.collect(joining(","));
-
-		// Prepare proxy to enter the cluster
-		final SocatContainer proxy = new SocatContainer().withNetwork(network);
-		configuration.iterateCoreMembers()
-			.forEach(member -> proxy.withTarget(DEFAULT_BOLT_PORT + member.getKey(), member.getValue(), DEFAULT_BOLT_PORT));
-
-		// Start the proxy so that the exposed ports are available and we can get the mapped ones
-		proxy.start();
-
-		final boolean is35 = configuration.getNeo4jVersion().startsWith("3.5");
-		// Build the cluster
-		final List<Neo4jContainer> cluster = configuration.iterateCoreMembers()
-			.map(member -> new Neo4jContainer<>(configuration.getImageName())
-				.withEnv("NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes")
-				.withAdminPassword(configuration.getPassword())
-				.withNetwork(network)
-				.withNetworkAliases(member.getValue())
-				.withCreateContainerCmdModifier(cmd -> cmd.withHostName(member.getValue()))
-				.withNeo4jConfig("dbms.mode", "CORE")
-				.withNeo4jConfig("dbms.memory.pagecache.size", configuration.getPagecacheSize() + "M")
-				.withNeo4jConfig("dbms.memory.heap.initial_size", configuration.getInitialHeapSize() + "M")
-				.withNeo4jConfig(is35 ? "dbms.connectors.default_listen_address" : "dbms.default_listen_address", "0.0.0.0")
-				.withNeo4jConfig(is35 ? "dbms.connectors.default_advertised_address" : "dbms.default_advertised_address", member.getValue())
-				.withNeo4jConfig("dbms.connector.bolt.advertised_address", String.format("%s:%d", proxy.getContainerIpAddress(), proxy.getMappedPort(DEFAULT_BOLT_PORT + member.getKey())))
-				.withNeo4jConfig("causal_clustering.initial_discovery_members", initialDiscoveryMembers)
-				.withNeo4jConfig("causal_clustering.minimum_core_cluster_size_at_formation", Integer.toString(numberOfCoreMembers))
-				.withNeo4jConfig("causal_clustering.minimum_core_cluster_size_at_runtime", Integer.toString(numberOfCoreMembers))
-				.withStartupTimeout(configuration.getStartupTimeout()))
-			.collect(toList());
-
-		// Start all of them in parallel
-		final CountDownLatch latch = new CountDownLatch(numberOfCoreMembers);
-		cluster.forEach(instance -> CompletableFuture.runAsync(() -> {
-			instance.start();
-			latch.countDown();
-		}));
-
-		try {
-			latch.await();
-		} catch (InterruptedException e) {
-			e.printStackTrace();
-		}
-
-		return new CausalCluster(proxy, cluster);
-	}
-
 	private final SocatContainer proxy;
```

Comment:
If we introduce the `boltPort`, this should also be named `boltProxy`.

Reply:
I actually do that in a subsequent chang, when I introduce a clusterProxy for cluster comms, easy to do it now though :-D

Reply:
done